ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Types.hs

```diff
@@ -765,3 +876,20 @@ data TraceIteratorEvent blk
     -- next block we're looking for.
   | SwitchBackToVolatileDB
   deriving (Generic, Eq, Show)
+
+{-------------------------------------------------------------------------------
+  Auxiliary types
+-------------------------------------------------------------------------------}
+
+-- | A wrapper for which '==' always returns 'True'
+--
+-- Use this in a field in a data type whose invariants ensure some other field
+-- effectively identifies this field.
+newtype Ignorable (sym :: Symbol) x = Ignorable {getIgnorable :: x}
```

Comment:
I haven't given thought to where this type declaration should actually be defined. It's an annoying shortcoming of GHC Haskell that we need these. (We could get around it, but only if we established a fixed interface boundary on `Tracer` events, and used eg pattern synonyms to hide it. Which we haven't bothered with so far.)